From 5f418395f57d8820b08a2766c9a7d343ec6cdb9a Mon Sep 17 00:00:00 2001
From: Sai Surya Teja <sssteja@habana.ai>
Date: Fri, 29 Oct 2021 11:28:24 +0300
Subject: [PATCH] [SW-59942] gaudi2: remove unwanted logs

Removed "Skipping halt for ARC" log in preboot.
Added log only if we need to halt ARC.
This is to make sure we are not cluttering the boot up logs
in gaudi2

Change-Id: I8a8cd225f444a48991af4913c057f7f32fbee003
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 4b4bceb..5932a69 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -929,10 +929,9 @@
 
 	halt_ack_offset = preboot_get_halt_offset(id);
 	val = regmap_readl(halt_ack_offset);
-	if (val & halt_mask) {
-		preboot_puts_hex("Skipping halt for ARC", id, false);
-	} else {
-		/* Halt ARC */
+	if (!(val & halt_mask)) {
+		/* Halt ARC incase if it is not already */
+		preboot_puts_hex("Halting ARC", id, false);
 		regmap_writel(BIT(1), halt_req_offset);
 		while (1) {
 			val = regmap_readl(halt_ack_offset);
